### PR TITLE
Auth: Disable team sync for JWT Authentication

### PR DESCRIPTION
**Pull request by eleijonmarck**
**What is this feature?**
Grafana Group/Team Sync runs for JWT auth when it shouldn't as JWT doesn't support external groups. Some users that sign in via OAuth with external group sync with group mappings. When the same users auth via JWT for API usage, group sync removes their group mappings and any permissions granted via groups.

This feature disables team synchronization, both for the JWT authn broker path and the JWT OAuth path.

**Which issue(s) does this PR fix?**:
https://github.com/grafana/grafana/issues/62175

Fixes #

**Special notes for your reviewer**:

### CODE REVIEW

**Comment by Jguer** [comment]
Seems to conform to the expected behavior

File: pkg/services/login/loginservice/loginservice_test.go [hunk 1]
```diff
@@ -84,7 +84,7 @@ func Test_teamSync(t *testing.T) {
 	var actualUser *user.User
 	var actualExternalUser *models.ExternalUserInfo
 
-	t.Run("login.TeamSync should not be called when  nil", func(t *testing.T) {
+	t.Run("login.TeamSync should not be called when nil", func(t *testing.T) {
```
Comment (eleijonmarck) [inline pkg/services/login/loginservice/loginservice_test.go:87]: not part of PR: just a fix for too many whitespaces

File: pkg/services/login/loginservice/loginservice_test.go [hunk 2]
```diff
@@ -103,6 +103,29 @@ func Test_teamSync(t *testing.T) {
 			assert.Equal(t, actualExternalUser, upserCmd.ExternalUser)
 		})
 
+		t.Run("login.TeamSync should not be called when not nil and skipTeamSync is set for externalUserInfo", func(t *testing.T) {
+			var actualUser *user.User
+			var actualExternalUser *models.ExternalUserInfo
+			upserCmdSkipTeamSync := &models.UpsertUserCommand{
```
Comment (linoman) [inline pkg/services/login/loginservice/loginservice_test.go:109]:
```suggestion
			upsertCmdSkipTeamSync := &models.UpsertUserCommand{
```

Comment (linoman) [reply pkg/services/login/loginservice/loginservice_test.go]: Looks good now. Thanks!

**Review by linoman — APPROVED** [review]
Great work!

**Comment by eleijonmarck** [comment]
setting up the oauth devenv together with generating a jwt to access the Grafana APIs. I was able to replicate to skip the team sync.
devenv/docker/blocks/auth/oauth/readme.md


So it is tested in a env where we reach the code path from API usage

**Pull request merged**